dotenv/src/parse.rs

```diff
@@ -173,7 +173,7 @@ fn parse_value(input: &str, substitution_data: &mut HashMap<String, Option<Strin
 
     //XXX also fail if escaped? or...
     if substitution_mode == SubstitutionMode::EscapedBlock || strong_quote || weak_quote {
-        Err(Error::LineParse(input.to_owned(), input.len() - 1))
+        Err(Error::LineParse(input.to_owned(), (input.len() - 1).max(0)))
```

Comment:
This doesn't fix the overflow case. When `len()` is 0 then `len() - 1` will be `usize::max` which is certainly greater than 0.